client/dfclient/src/menu_state.cpp

```diff
@@ -91,8 +92,8 @@ MenuState::MenuState(Resources& r) : _resources(r) {
 void MenuState::TryConnect() {
 	gameData.serverAddress = "ws://" + gameData.serverAddress;
 	std::cout << "server " << gameData.serverAddress << ", my nickname " << gameData.myNickname << "\n";
-	gameData.socket = CreateWebSocket();
-	int ret = gameData.socket->Connect(gameData.serverAddress);
+	CreateWebSocket();
+	int ret = webSocketManager._ws->Connect(gameData.serverAddress);
 	if (ret < 0) {
```

Comment:
The race is here - adding `sleep(100ms)` after `Connect()` seemed to solve the problem.

Reply:
Please use proper synchronization to fix it, though (e.g. mutexes, condition variables, semaphores).

Reply:
@piodul Don't get me wrong, it was just a note to pinpoint the problem, not a fix suggestion xD

Reply:
According to my research the race should not be present, even though I did nothing in the code to mitigate it. After the call to `handshake` finished successfully the socket should be immediately available to send data. The only thing that might be racing here is if there is something happening within `ioc.run` that makes the websocket ready to send messages and there is no way to synchronize anything within an external function call except for a `sleep`.
After some testing I have **not** observed the race happening on my machine. It is possible that the racing behaviour was introduced in a different PR.